Allow top diffs to use 30 seconds of break time for spread rules in `Ranking criteria` (#8861)
ppy/osu-wiki: wiki/Ranking_Criteria/en.md

  - **...lower than 3:30**, the lowest difficulty of each included game mode cannot be harder than a Normal.
  - **...between 3:30 and 4:15**, the lowest difficulty of each included game mode cannot be harder than a Hard.
  - **...between 4:15 and 5:00**, the lowest difficulty of each included game mode cannot be harder than an Insane.

>   - **[Break times](/wiki/Beatmap/Break) may be combined with [drain time](/wiki/Beatmap/Drain_time) to meet the above thresholds.** For the highest difficulty, this is limited to at most 30 seconds of break time. This does not apply to difficulties with less than 30 seconds of drain time.

Reviewer: This latest version still doesn't address my concerns [here](https://github.com/ppy/osu-wiki/pull/8861#discussion_r1119932473). The wording of this rule means it would be valid to make an easy difficulty with 30s drain time and 5 minutes break time.

Author: it has always been like this and has never been an issue, so I'm afraid that's out of scope for now

Reviewer: Okay. I'm still very unhappy with the wording but if you guys think this is the correct path then let's see.

>   - **Difficulties below the highest difficulty can combine [break times](/wiki/Beatmap/Break) with [drain time](/wiki/Beatmap/Drain_time) to meet the above thresholds.** The highest difficulty can combine up to 30 seconds worth of break times with drain time instead. These do not apply to difficulties with less than 30 seconds of drain time.

Reviewer: Can someone explain the implication of these changes? I'm not sure if it's the way this is written / structured but my brain can't comprehend.

Is it basically saying that "break time can now be used to bad a beatmap up to the required length to mean that the mapper can map less difficulties"?

Author: yes that is what it means, and they already are based on drain time. highest difficulty has always been the only one that wasn't allowed to use break time to meet the requirements for... some reason. so we're trying to relax that, but people are afraid there will be abuse if we don't set some sane limit on how many breaks you can have so you don't have a 5 minute map with 3 minutes of breaks in ranked

id argue bns would know better (and it would be much less confusing without the arbitrary 30 seconds restriction) but this solution seemed to please everyone

Reviewer: I don't know. I think to a mapper this whole criteria is so confusing the majority would not be able to understand it. So let's brainstorm a better way to go about this.

Something like changing the base time used here from `drain_time` to `drain_time + 30s of break time` or just leaving it as `drain_time` but adjusting the ranges??

Reviewer 2: @peppy just so we're on the same page, i've simplified this bullet point in https://github.com/ppy/osu-wiki/pull/8861/files#diff-d83e7a1fb8077f937a9a91827c6cb673767a7ebb721e3482bdc146a80802b3d2R63

>   - **[Break times](/wiki/Beatmap/Break) may be combined with [drain time](/wiki/Beatmap/Drain_time) to meet the above thresholds.** For the highest difficulty, this is limited to at most 30 seconds of break time. This does not apply to difficulties with less than 30 seconds of drain time.

i don't think this is at all hard to understand now.

Reviewer: Personally I still think this is the wrong way to go about it (for instance, how much does this differ from adjusting each time value in the range criteria downwards by 30s?).

And it could still be abused by making an ENHI split and making the easy 99% break time, so I'm not sure what the 30s part is fixing here. It should be applied across the board if it is to close this loophole.

>   - **Difficulties below the highest difficulty can combine [break times](/wiki/Beatmap/Break) with [drain time](/wiki/Beatmap/Drain_time) to meet the above thresholds.** The highest difficulty can combine up to 30 seconds worth of break times with drain time instead. These do not apply to difficulties with less than 30 seconds of drain time.

Reviewer 2: ```suggestion
  - **[Break times](/wiki/Beatmap/Break) may be combined with [drain time](/wiki/Beatmap/Drain_time) to meet the above thresholds.** For the highest difficulty, this is limited to at most 30 seconds of break time. This does not apply to difficulties with less than 30 seconds of drain time.
```
